update IEEE 802.15.4 driver

Cristib05 opened the pull request:
drivers: ieee802154: updated mcxw ieee802154 driver
samples: echo_client echo_server: update tests
doc: frdm_mcxw72: update documentation
modules: hal_nxp: update commit to PR 547

Testing
Compiled echo_server/echo_client with: west build -p always -bfrdm_mcxw72/mcxw727c/cpu0 samples/net/sockets/echo_client -- -DEXTRA_CONF_FILE='overlay-ot.conf'
Created a thread network with it and joined another thread device
Checked that the thread communication is working (with a 15.4 thread wireshark capture)
Was able to see the neighbors and ping

Cristib05 committed:
doc: frdm_mcxw72: update documentation

Update documentation for dynamic NBU.
Add OpenThread application building examples.

Signed-off-by: Cristian Bulacu <cristian.bulacu@nxp.com>

Cristib05 committed:
samples: echo_client echo_server: update tests

Added frdm_mcxw72 as allowed platform

Signed-off-by: Cristian Bulacu <cristian.bulacu@nxp.com>

Cristib05 committed:
drivers: ieee802154: update mcxw ieee802154 driver

Updated driver based on changes done in SDK 25.06.00-pvw2.
Update hal_nxp to include BLE LL and IEEE 802.15.4 PHY NBU
combo firmware for MCXW72 boards.

Signed-off-by: Cristian Bulacu <cristian.bulacu@nxp.com>

jukkar reviewed (dismissed):
LGTM

rlubos reviewed (commented):
Looks good (provided manifest is updated to use proper SHA).

Cristib05 committed:
drivers: ieee802154: Fix resolution in mcxw_get_time function.

Updated mcxw_get_time function to output its result in nanosecond
resolution and updated code accordingly.

Signed-off-by: Cristian Bulacu <cristian.bulacu@nxp.com>

dleach02 commented:
close/open to clear module block

George-Stefan commented:
@rlubos since you're one of the assignees, can you please merge this PR

rlubos commented:
I don't have merge rights.

dkalowsk commented:
@Cristib05 I'm seeing conflicts on the merge.  Please resolve so we can move forward.

mmahadevan108 commented on west.yml:
```diff
@@ -210,7 +210,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: dcb80803ee9d528c600d012aa4ac6515bbd8fb3d
```
Change this to pull in only your commit.
